Firestore/core/src/firebase/firestore/immutable/array_sorted_map.h

```diff
@@ -334,6 +333,16 @@ class ArraySortedMap : public SortedMapBase {
     return kEmptyArray;
   }
 
+  static array_pointer SortedArray(std::initializer_list<value_type> entries,
+                                   const C& comparator) {
+    std::vector<value_type> sorted{entries.begin(), entries.end()};
```

Comment:
Rather than creating (yet another) copy, we could instead just create the FixedArray, and then sort it in place. However, our ArraySortedMap has a const reference to the FixedArray (which I'd like to keep const) and it seems unlikely we'd every have a large enough initializer_list to matter.

Another alternative would be to have a FixedArray ctor that also happens to sort. But FixedArray currently has no concept of sorting, and I'd rather not introduce it.

Reply:
Optional: an alternative could be simply asserting that the initializer list given is sorted:
```cpp
HARD_ASSERT(std::is_sorted(entries.begin(), entries.end(), comparator), "The initializer list must be in sorted order");
```
and fixing all the callers. However, I have no idea how many such callers there are, so this might be infeasible to do.

Reply:
That might be reasonable. However, the reason I discovered this bug in the first place, is because we're now (or will shortly be) using this as the backing for FieldValue::Map, which doesn't ask for it's input to be sorted. Changing all the call sites of FieldValue::Map(initializers) across all three repos isn't something I'm exited about doing. (The overwhelming majority of the non-trivial uses are from the tests, where the order might be purposeful too.)

OTOH, I *could* move the sort to the fieldvalue::map ctor and just assert here as you've indicated. It won't avoid sorting though.

FWIW, the android implementation also copies and sorts it's input, so I'm still leaning toward just sorting here too.

Reply:
Ah, I see. In that case, sorting sounds like the right thing to do. Also, I presume most (all?) uses of this constructor are in tests, where the efficiency argument just doesn't apply.

Reply:
AFAICT, that's correct (excepting a couple of empty, i.e. `{}` initializers.)